core/src/test/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraintsTest.java

```diff
@@ -0,0 +1,84 @@
+package fr.sncf.osrd.api.pathfinding.constraints;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.google.common.collect.Range;
+import fr.sncf.osrd.api.pathfinding.PathfindingRoutesEndpoint;
+import fr.sncf.osrd.api.pathfinding.response.NoPathFoundError;
+import fr.sncf.osrd.infra.api.Direction;
+import fr.sncf.osrd.stdcm.DummyRouteGraphBuilder;
+import fr.sncf.osrd.train.TestTrains;
+import fr.sncf.osrd.utils.graph.Pathfinding;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class ElectrificationConstraintsTest {
+    static Stream<Arguments> testDeadSectionArgs() {
+        return Stream.of(
+                // How much catenary, with dead section, dead section direction, should block some ranges
+                Arguments.of(2, true, Direction.FORWARD, false),
+                Arguments.of(2, true, Direction.BACKWARD, false),
+                Arguments.of(2, false, Direction.FORWARD, false),
+                Arguments.of(1, true, Direction.FORWARD, false),
+                Arguments.of(1, true, Direction.BACKWARD, true),
+                Arguments.of(1, false, Direction.FORWARD, true),
+                Arguments.of(0, true, Direction.FORWARD, true),
+                Arguments.of(0, false, Direction.FORWARD, true)
+            );
+    }
+
+    @ParameterizedTest
+    @MethodSource("testDeadSectionArgs")
+    public void testDeadSectionAndCatenaryBlockingRanges(int howMuchCatenary, boolean withDeadSection,
+                                                         Direction deadSectionDirection, boolean shouldBlock)
+            throws IOException, URISyntaxException {
+
+        var route = new DummyRouteGraphBuilder.DummyRoute("route", 1000, null, null, null);
+
+        assert TestTrains.FAST_ELECTRIC_TRAIN.getModeNames().contains("25000");
+        switch (howMuchCatenary) {
+            case 0:
+                break;
+            case 1:
+                route.getTrackRanges().get(0).track.getEdge().getVoltages().put(Range.closed(0., 100.), "25000");
+                route.getTrackRanges().get(0).track.getEdge().getVoltages().put(Range.closed(900., 1000.), "25000");
+                break;
+            case 2:
+                route.getTrackRanges().get(0).track.getEdge().getVoltages().put(Range.closed(0., 1000.), "25000");
+                break;
+            default:
+                throw new IllegalArgumentException("howMuchCatenary must be 0, 1 or 2");
+        }
+
+        if (withDeadSection) {
+            route.getTrackRanges().get(0).track.getEdge().getDeadSections(deadSectionDirection)
+                    .add(Range.closed(100., 900.));
+        }
+
+        var blockedRanges = ElectrificationConstraints.getBlockedRanges(TestTrains.FAST_ELECTRIC_TRAIN, route);
```

Comment:
```suggestion
        var blockedRanges = new ElectrificationConstraints(List.of(TestTrains.FAST_ELECTRIC_TRAIN)).apply(route);
```

That way `getBlockedRanges` can remain private

Reply:
I don't understand why `getBlockedRanges` being private and not package private is important

Reply:
It's best to test through the public interfaces (when reasonable) and leave the rest as restricted as possible. But it's not a big deal either.